Lesson5/main.js

```diff
@@ -0,0 +1,24 @@
+const ul = document.getElementById("js-ul");
+const fragment = new DocumentFragment();
+const features = [{to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"},
+{to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}];
+
+const checkFeatures = new Promise((resolve) => {
+  resolve(features);
+}).then(
+  function(value){
+    for (let feature of features ){
+      const li = document.createElement('li');
+      const a = document.createElement('a');
+      const img = document.createElement('img');
+      a.href = feature.to;
+      a.textContent = feature.text;
+      img.src = feature.img;
+      img.alt = feature.alt
+
+      fragment.appendChild(li).appendChild(a).insertAdjacentElement('afterbegin', img);
+    }
+
+    ul.appendChild(fragment);
+  }
+);
```

Comment:
Comment
I think Saki-san's way is fine!
It's great that Promise seems to have figured it out so quickly!

It is also possible to write it in an Arrow function as follows.
However, I think either is fine.

I often use the Arrow function because it requires fewer lines to write.
However, the Arrow function does not have 'this', so care must be taken if 'this' is required.

```suggestion
}).then((value) => {
    for (let feature of features ){
      const li = document.createElement('li');
      const a = document.createElement('a');
      const img = document.createElement('img');
      a.href = feature.to;
      a.textContent = feature.text;
      img.src = feature.img;
      img.alt = feature.alt

      fragment.appendChild(li).appendChild(a).insertAdjacentElement('afterbegin', img);
    }

    ul.appendChild(fragment);
  });
```